### PR TITLE
[Reviewer: Richard] Add Chronos config plugins and scripts

### PR DESCRIPTION
**Pull request by eleanor-merry**
Adds the Chronos config plugins and scripts.

Tested in UTs and live

### CODE REVIEW

File: chronos/scripts/force_chronos_gr_restart_queue_state [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+local_site_name=site1
+. /etc/clearwater/config
+
+if [ $# -ne 0 ]
+then
+  echo "Usage: force_chronos_gr_restart_queue_state"
+  exit 1
+fi
+
+/usr/share/clearwater/clearwater-queue-manager/env/bin/python /usr/share/clearwater/clearwater-queue-manager/scripts/force_queue_state.py "${management_local_ip:-$local_ip}" "$local_site_name" "apply_chronos_gr_config"
```
Comment (richardwhiuk) [inline chronos/scripts/force_chronos_gr_restart_queue_state]: You can split long lines in bash by using a backslash at the end of the line:

``` bash
/usr/share/clearwater/clearwater-queue-manager/env/bin/python \
  /usr/share/clearwater/clearwater-queue-manager/scripts/force_queue_state.py \
    "${management_local_ip:-$local_ip}" \
    "$local_site_name" \
    "apply_chronos_gr_config"
```

File: chronos/scripts/check_chronos_gr_restart_queue_state [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+local_site_name=site1
+. /etc/clearwater/config
+
+if [ $# -ne 0 ]
+then
+  echo "Usage: check_chronos_gr_restart_queue_state"
+  exit 1
+fi
+
+/usr/share/clearwater/clearwater-queue-manager/env/bin/python /usr/share/clearwater/clearwater-queue-manager/scripts/check_queue_state.py "${management_local_ip:-$local_ip}" "$local_site_name" "apply_chronos_gr_config"
```
Comment (richardwhiuk) [inline chronos/scripts/check_chronos_gr_restart_queue_state]: Can you split this line up so it's more readable?

File: chronos/scripts/upload_chronos_gr_config [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# Check the return code and log if appropriate.
+if [ $rc != 0 ] ; then
+  echo Upload chronos GR configuration failed to $keypath  >&2
+  cat /tmp/upload-chronos-gr-config.stderr.$$              >&2
+  cat /tmp/upload-chronos-gr-config.stdout.$$              >&2
+fi
+rm -f /tmp/upload-chronos-gr-config.stderr.$$ /tmp/upload-chronos-gr-config.stdout.$$
+
+# Add this node to the list of nodes to restart
+/usr/share/clearwater/clearwater-queue-manager/scripts/modify_nodes_in_queue add apply_chronos_gr_config
```
Comment (richardwhiuk) [inline chronos/scripts/upload_chronos_gr_config]: If there any point restarting this node if the config failed to upload to etcd?

File: chronos/apply_chronos_gr_config_plugin.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                "global")
+
+    def key(self): # pragma: no cover
+        return "apply_chronos_gr_config"
+
+    def at_front_of_queue(self):
+        _log.info("Restarting Chronos")
+        run_command("service chronos stop")
+        run_command("service chronos wait-sync")
+        run_command("/usr/share/clearwater/clearwater-queue-manager/scripts/modify_nodes_in_queue remove_success apply_chronos_gr_config")
```
Comment (richardwhiuk) [inline chronos/apply_chronos_gr_config_plugin.py]: If some of these commands fail, run_command will return the error code, but we'll continue anyway - which sounds dangerous.

File: chronos/chronos_gr_config_plugin.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    def status(self, value):
+        try:
+            with open(_file, "r") as ifile:
+                current = ifile.read()
+                if current == value:  # pragma: no cover
+                    return FileStatus.UP_TO_DATE
+                else:
+                    return FileStatus.OUT_OF_SYNC
+        except IOError:  # pragma: no cover
+            return FileStatus.MISSING
```
Comment (richardwhiuk) [inline chronos/chronos_gr_config_plugin.py:62]: This looks copy and pasted from SproutJsonPlugin (or possibly elsewhere). Can we move this into a common method?

File: chronos/scripts/upload_chronos_gr_config [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+curl -X PUT $keypath --data-urlencode value@$FILENAME 2> /tmp/upload-chronos-gr-config.stderr.$$ | tee /tmp/upload-chronos-gr-config.stdout.$$ | egrep -q "\"action\":\"set\""
+rc=$?
+
+# Check the return code and log if appropriate.
+if [ $rc != 0 ] ; then
+  echo Upload chronos GR configuration failed to $keypath  >&2
+  cat /tmp/upload-chronos-gr-config.stderr.$$              >&2
+  cat /tmp/upload-chronos-gr-config.stdout.$$              >&2
+fi
+rm -f /tmp/upload-chronos-gr-config.stderr.$$ /tmp/upload-chronos-gr-config.stdout.$$
```
Comment (richardwhiuk) [inline chronos/scripts/upload_chronos_gr_config:71]: It feels like the lines above should be something that would be in a common function... in a sensible language (i.e. not bash).

**Comment by richardwhiuk** [comment]
Looks good. Some common function we should commonalize, and a couple of stylistic comments.

File: chronos/apply_chronos_gr_config_plugin.py (hunk 4 above)
Comment (eleanor-merry) [reply chronos/apply_chronos_gr_config_plugin.py]: Continuing is the right choice in this case - I'll add some logs though to make it clear when this has happened

**Comment by eleanor-merry** [comment]
Closing this PR for now

**Pull request closed without merging**